werkzeug/wrappers.py

```diff
@@ -789,6 +789,15 @@ def __init__(self, response=None, status=None, headers=None,
         else:
             self.response = response
 
+    def __set_response(self, value):
+        self._cached_data = None
+        self.__response = value
+
+    def __get_response(self):
+        return self.__response
+
+    response = property(__get_response, __set_response)
```

Comment:
It is possible to change data with assign `response`. This property just drop the `_cached_data`.